File: fetch_projects.py
import json
import urllib.request
import os

env = {}
with open('.env', 'r') as f:
    for line in f:
        if '=' in line:
            k, v = line.strip().split('=', 1)
            env[k] = v.strip('"\'')

url = env['SUPABASE_URL'] + '/rest/v1/projects?select=id,slug,title_en,title_ar'
req = urllib.request.Request(url, headers={
    'apikey': env['SUPABASE_PUBLISHABLE_KEY'],
    'Authorization': 'Bearer ' + env['SUPABASE_PUBLISHABLE_KEY']
})

try:
    res = urllib.request.urlopen(req).read().decode('utf-8')
    with open('supabase_projects.json', 'w', encoding='utf-8') as f:
        f.write(res)
    print("Success")
except Exception as e:
    print("Error:", e)

File: fetch_projects.js
import { createClient } from "@supabase/supabase-js";
import fs from "fs";

const envText = fs.readFileSync(".env", "utf-8");
const env = {};
for (const line of envText.split("\n")) {
  if (line.includes("=")) {
    const [k, v] = line.split("=");
    env[k.trim()] = v.trim().replace(/^["']|["']$/g, "");
  }
}

const supabase = createClient(env["SUPABASE_URL"], env["SUPABASE_PUBLISHABLE_KEY"]);

async function run() {
  const { data, error } = await supabase
    .from("vw_public_projects")
    .select("project_id, slug, title, locale");
  if (error) {
    console.error(error);
  } else {
    fs.writeFileSync("supabase_projects.json", JSON.stringify(data, null, 2));
    console.log("Success", data.length, "projects fetched.");
  }
}

run();
